### PR TITLE
feat: add high level parameters to dashboard that can be passed to charts/queries

### PR DESCRIPTION
**Pull request by glinton**
**TL;DR** Shelving this implementation for a bit.

---------------------------------

The following template defines individual parameters per chart query. My though was to add the ability to define dashboard-wide parameters intended for use inside chart queries to limit the number of unique definitions required for the parameters.

**Original parameterized template:**
```yml

apiVersion: influxdata.com/v2alpha1
kind: Dashboard
metadata:
    name: linux-system
spec:
    name: Linux System
    charts:
        queries:
          - query: |-
                option params = {
                  bucket: "v.bucket"
                }

                from(bucket: params.bucket)
                  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
                  |> filter(fn: (r) => r._measurement == "system")
                  |> filter(fn: (r) => r._field == "uptime")
                  |> last()
                  |> map(fn: (r) => ({ _value: float(v: r._value) / 86400.00 }))
        suffix: ' days'
```

**Diff of new functionality:**
```diff
7a8,11
>     params:
>       - key: bucket
>         default: "v.bucket"
>         type: string
11,14d14
<                 option params = {
<                   bucket: "v.bucket"
<                 }
< 
```

This idea falls apart when implementing, flux complains that params is not defined, suggesting that the fix is to export the template with the parent params injected into each child query... which would then bring us full circle to needing to define each parameter in each query.

---------------------------

Another implementation idea is to parameterize a variable, then reference that variable in the dashboard chart queries where it is intended to be re-used in a set-once, use-many setting.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [ ] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [ ] Rebased/mergeable
- [ ] Tests pass
- [ ] http/swagger.yml updated (if modified Go structs or API)
- [ ] Feature flagged (if modified API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by lesam** [comment]
@glinton can we close this?
